worker-sys/src/types/durable_object/state.rs

```diff
@@ -15,4 +15,23 @@ extern "C" {
 
     #[wasm_bindgen(method, js_name=waitUntil)]
     pub fn wait_until(this: &DurableObjectState, promise: &js_sys::Promise);
+
+    #[wasm_bindgen(method, js_name=acceptWebSocket)]
+    pub fn accept_web_socket(this: &DurableObjectState, ws: web_sys::WebSocket);
+
+    #[wasm_bindgen(method, js_name=acceptWebSocket)]
+    pub fn accept_web_socket_with_tags(
+        this: &DurableObjectState,
+        ws: web_sys::WebSocket,
+        tags: Vec<String>,
```

Comment:
The use of `Vec` instead of `&[&str]` was based on something I read in the `wasm-bindgen` docs about unboxed slices not being well supported. Although I can't seem to find that in the docs again so perhaps I'm misremembering. I'll try it out tomorrow and report back.

Reply:
Really? I'm getting the following error `the trait bound \`&[&str]: wasm_bindgen::convert::IntoWasmAbi\` is not satisfied`

Reply:
Everywhere else in the API where something similar is used, such as `get_multiple` in the `DurableObjectTransaction` uses `Vec<JsValue>`.

```rs
pub fn get_multiple(
        this: &DurableObjectTransaction,
        keys: Vec<JsValue>,
    ) -> Result<js_sys::Promise, JsValue>;
```

Reply:
Although `Vec<String>` is also giving me issues now, perhaps there was a change between `wasm-bindgen` 0.2.87 and 0.2.89. The problem only appeared after removing the changes I made to the `wasm-bindgen` version.

Reply:
Hey @eric-seppanen can you double check that this is matches your experience?

wasm_bindgen | Vec\<JsValue> | &[JsValue] | Vec\<String> | &[&str]
--- | --- | --- | --- | ---
v0.2.87 | :heavy_check_mark: | :x: | :x: | :x: 
v0.2.89 | :heavy_check_mark: | :x: | :heavy_check_mark:  | :x: 
v0.2.90 | :heavy_check_mark: | :x: | :heavy_check_mark:  | :x:

Reply:
We can definitely still have the public api's in `worker` accept `&[&str]` but they'll have to transform it to `Vec<String>` or `Vec<JsValue>` before calling the methods defined in `worker-sys`.

Reply:
Yeah, sorry, I misspoke.

What I did was this:
```rust
#[wasm_bindgen(method, js_name=acceptWebSocket)]
pub fn accept_websocket(this: &DurableObjectState, ws: &web_sys::WebSocket, tags: &js_sys::Object);
```
and
```rust
pub fn accept_websocket(&self, websocket: &WebSocket, tags: &[&str]) {
    let js_tags = js_sys::Array::new();
    for tag in tags {
        js_tags.push(&js_sys::JsString::from(*tag));
    }
    self.inner.accept_websocket(websocket.as_ref(), &js_tags)
}
```
So the slice reference was only on the user-facing part.